internal/util/update/resource-merge.go

```diff
@@ -131,6 +135,69 @@ func (u ResourceMergeUpdater) updatedKptfile(updatedPath, originalPath string, o
 	return updatedKf, err
 }
 
+// MergeSubPackages merges the Kptfiles in the nested subdirectories of the
+// root package and also sets the setter definitions in updated to match with
+// locally set values so the the resources are correctly identified and merged
+func MergeSubPackages(localRoot, updatedRoot, originalRoot string) error {
+	localPkgPaths, err := pathutil.DirsWithFile(localRoot, kptfile.KptFileName, true)
+	if err != nil {
+		return err
+	}
+	for _, localPkgPath := range localPkgPaths {
+		// skip the top level file as it should be merged differently
+		if filepath.Clean(localPkgPath) == filepath.Clean(localRoot) {
+			continue
+		}
+
+		relativePkgPath := strings.TrimPrefix(localPkgPath, localRoot)
+
+		localKf, err := kptfileutil.ReadFile(localPkgPath)
+		if err != nil {
+			return err
+		}
+
+		updatedPkgPath := filepath.Join(updatedRoot, relativePkgPath)
+		updatedKf, err := kptfileutil.ReadFile(updatedPkgPath)
+		if err != nil {
+			updatedKf = localKf
```

Comment:
This needs a comment.

Reply:
Added. The aim here is to fall back to local if there is no Kptfile in updated path so that the local Kptfile is retained.